javaslang/src/main/java/javaslang/collection/LinkedHashSet.java

```diff
@@ -846,7 +852,13 @@ public String mkString(CharSequence prefix, CharSequence delimiter, CharSequence
 
     @Override
     public LinkedHashSet<Tuple2<T, Long>> zipWithIndex() {
-        return LinkedHashSet.ofAll(iterator().zipWithIndex());
+        return zipWithIndex(Tuple::of);
+    }
+
+    @Override
+    public <U> LinkedHashSet<U> zipWithIndex(BiFunction<? super T, ? super Long, ? extends U> mapper) {
```

Comment:
It's like returning an `LinkedList`, the user of the code shouldn't care whether we change it to `ArrayList`, they should only see `List` in the public interface.
The same applies here, we should probably return `Set`, so that we can freely modify the underlying structure later.